babel/numbers.py

```diff
@@ -624,6 +625,36 @@ def _format_currency_long_name(
     return unit_pattern.format(number_part, display_name)
 
 
+def format_compact_currency(number, currency, *, format_type="short", locale=LC_NUMERIC, fraction_digits=0):
+    u"""Format a number as a compact currency value.
+
+    >>> format_compact_currency(12345, 'USD', locale='en_US')
+    u'$12K'
+    >>> format_compact_currency(123456789, 'USD', locale='en_US', fraction_digits=2)
+    u'$123.46M'
+    >>> format_compact_currency(123456789, 'EUR', locale='de_DE', fraction_digits=1)
+    u"123,5\xa0Mio'.'\xa0€"
+
+    :param number: the number to format
+    :param currency: the currency code
+    :param format_type: the compact format type to use. Defaults to "short".
+    :param locale: the `Locale` object or locale identifier
+    :param fraction_digits: Number of digits after the decimal point to use. Defaults to `0`.
+    """
+    locale = Locale.parse(locale)
+    try:
+        compact_format = locale.compact_currency_formats[format_type]
+    except KeyError as e:
+        raise UnknownCurrencyFormatError(
+            "%r is not a known currency format type" % format_type) from e
```

Comment:
You can use an f-string in new code.

```suggestion
        raise UnknownCurrencyFormatError(f"{format_type!r} is not a known currency format type") from e
```